torch_xla_py/xla_dist.py

```diff
@@ -78,38 +114,157 @@ def validate(self):
 
     if self._check_client_machine_type:
       client_machine_types = {
-          worker._machine_type for worker in self._client_workers}
+          worker._machine_type for worker in self._client_workers
+      }
       if len(client_machine_types) != 1:
         raise RuntimeError(
-            'All client_workers must have the same machine_type, got: {}'.format(
-                client_machine_types))
+            'All client_workers must have the same machine_type, got: {}'
+            .format(client_machine_types))
 
     if self._check_service_machine_type:
       server_machine_types = {
-          worker._machine_type for worker in self._service_workers}
+          worker._machine_type for worker in self._service_workers
+      }
       if len(server_machine_types) != 1:
         raise RuntimeError(
-            'All service_workers must have the same machine_type, got: {}'.format(
-                server_machine_types))
+            'All service_workers must have the same machine_type, got: {}'
+            .format(server_machine_types))
 
 
 class ClusterResolver(object):
   """Cluster Resolver for Client VM and Cloud TPU mesh."""
 
-  def __init__(self,
-               tpus,
-               vms=None,
-               zone=None,
-               project=None):
+  @classmethod
```

Comment:
I am curious: why use @classmethod here? Is it to make mocking easier when testing?

Reply:
Since this method doesn't use anything bound to the object instantiation itself, I thought making it a `@classmethod` would be more apt (could be called outside if needed).

Reply:
It seems you don't use the class object either, so I think you could use @staticmethod instead of @classmethod unless that interferes with mocking/testing somehow

Reply:
Done. Thanks.